contracts/validator-manager/SlotAuctionManager.sol

```diff
@@ -5,242 +5,440 @@
+        /// @notice The minimum amount of time a validator is allowed to validate
+        uint256 _minValidatorDuration;
+        /// @notice The minimum amount of time the auction must run
+        uint256 _minAuctionDuration;
+        /// @notice The minimum bid to be placed into the auction
+        uint256 _minimumBid;
+        /// @notice The amount of time until the next auciton can be initiated
+        uint256 _auctionCooldownEndtime;
+        /// @notice Maps nodeIDs to boolean to check if nodeID is already in the running and capable of winning
+        mapping(bytes nodeID => bool isQualified) _nodeIsQualified;
+        /// @notice Maps nodeIDs to a struct holding validator information
+        mapping(bytes nodeID => ValidatorInfo) _validatorsByNodeID;
+        /// @notice Maps ValidationIDs to a struct holding validator information
+        mapping(bytes32 validationID => ValidatorInfo) _validatorsByValidationID;
+        /// @notice Maps bid amount to a struct containing the bidders info
+        mapping(uint256 bid => ValidatorBid) _bidderInfo;
+        /// @notice The current second price for the lowest slot winner, is 0 if no second price
+        uint256 _secondPrice;
+        /// @notice The data structure holding auction bid information
+        Heap.Uint256Heap _bids;
+    }
+    // solhint-enable private-vars-leading-underscore
+    // keccak256(abi.encode(uint256(keccak256("avalanche-icm.storage.SlotAuctionManager")) - 1)) & ~bytes32(uint256(0xff));
+
+    bytes32 public constant SLOT_AUCTION_MANAGER_STORAGE_LOCATION =
+        0x6494e1e6ac2d5f44a06b929c1549cec0c499347f244a852c37aef6b6707be600;
 
     error AuctionInProgress();
     error AuctionNotInProgress();
     error NodeIsValidator(bytes nodeID);
-    error NodeIsWinning(bytes nodeID);
+    error DuplicateNodeIDInContention(bytes nodeID);
     error DuplicateBidInContention(uint256 bid);
     error BidSmallerThanMinimum(uint256 minumumBid, uint256 userBid);
     error InsufficientBidToWinAuction(uint256 smallestAcceptableBid, uint256 userBid);
     error AuctionEndTimeNotReached(uint256 auctionEndTime);
     error ValidatorTimeLimitNotPassed(uint256 validationTimeLimit);
     error AuctionFinalizing();
+    error ValidatorWeightTooHigh(uint64 validatorWeight);
+    error NoOpenValidatorSlots(uint16 validatorSlots);
+    error MoreActiveValidatorsThanTotalSlots(uint16 totalValidatorSlots, uint16 activeValidators);
+    error AuctionInCooldown(uint256 auctionCooldownEndtime);
+    error InvalidWeight(uint64 weight);
+    error InvalidMinValidatorDuration(uint256 minimumValidationDuration);
+    error ZeroAddress();
+    error ZeroMinBid();
 
-    modifier AuctionOn {
-        if (auctionState != AuctionState.AuctionInProgress) {
+    // solhint-disable ordering
+
+    modifier AuctionOn() {
+        SlotAuctionManagerStorage storage $ = _getSlotAuctionManagerStorage();
+
+        if ($._auctionState != AuctionState.AuctionInProgress) {
```

Comment:
This modifier just checks that the auction is in progress (which is correct), but the other modifier checks for each of the other states and emits a different error for each. I don't have a strong preference on which way you do it, but it should be consistent.